src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -12,8 +12,8 @@
   and the jumpoff year. Returns a dataset where the column to be averaged contains
   the averages for all the years of interest and is renamed to avg-name."
   [historical-data col-to-avg avg-name start-yr-avg end-yr-avg]
-  ;; start-yr-range should be >= than the earliest year in historical-data
-  ;; end-yr-range should be <= than the latest year in historical-data
+  ;; (s/validate (s/pred (>= % historical-data-earliest-yr)) start-yr-range)
+  ;; (s/validate (s/pred (<= % historical-data-latest-yr)) end-yr-range)
```

Comment:
Sorry, I misunderstood you. I thought it wasn't to actually validate but to explain more clearly what I meant in the comments. I'll add a validation step now!

Is it ok to leave what I did here? https://github.com/MastodonC/witan.models.demography/blob/feature/start-end-yr-for-avg-proj/test/witan/models/dem/ccm/core/projection_loop_test.clj#L37-L51